RadixWallet/Features/AssetsFeature/Components/StakeUnitList/Components/StakeClaimNFTsView.swift

```diff
@@ -82,24 +65,80 @@ public struct StakeClaimNFTSView: View {
 				Spacer()
 			}
 
-			if !viewState.unstaking.isEmpty {
-				sectionView(viewState.unstaking, kind: .unstaking)
-			}
-
-			if !viewState.readyToBeClaimed.isEmpty {
-				sectionView(viewState.readyToBeClaimed, kind: .readyToBeClaimed)
-			}
-
-			if !viewState.toBeClaimed.isEmpty {
-				sectionView(viewState.toBeClaimed, kind: .toBeClaimed)
-			}
+			StakeClaimTokensView(
+				viewState: viewState.stakeClaimTokens,
+				background: background,
+				onTap: onTap,
+				onClaimAllTapped: onClaimAllTapped
+			)
 		}
 		.padding(.medium3)
 		.background(background)
 	}
+}
+
+// MARK: - StakeClaimTokensView
+public struct StakeClaimTokensView: View {
+	enum SectionKind {
+		case unstaking
+		case readyToBeClaimed
+		case toBeClaimed
+	}
+
+	public struct ViewState: Sendable, Hashable {
+		public let canClaimTokens: Bool
+		public let stakeClaims: IdentifiedArrayOf<OnLedgerEntitiesClient.StakeClaim>
+		var selectedStakeClaims: IdentifiedArrayOf<NonFungibleGlobalId>?
+
+		var unstaking: IdentifiedArrayOf<OnLedgerEntitiesClient.StakeClaim> {
+			stakeClaims.filter(\.isUnstaking)
+		}
+
+		var readyToBeClaimed: IdentifiedArrayOf<OnLedgerEntitiesClient.StakeClaim> {
+			stakeClaims.filter(\.isReadyToBeClaimed)
+		}
+
+		var toBeClaimed: IdentifiedArrayOf<OnLedgerEntitiesClient.StakeClaim> {
+			stakeClaims.filter(\.isToBeClaimed)
```

Comment:
the name `isToBeClaimed` is too similar to `isReadyToBeClaimed` IMO, thus is `readyToBeClaimed` IdentifiedArrayOf and `toBeClaimed` array variable names also too similar. does "isToBeClaimed" really mean "notYetReadyToBeClaimed"? Then I think we should name it that! Else, what is it? :D

Reply:
This is consistent with representation at the UI level, when two different strings are shown "Ready to be claimed" and "To be claimed", I did choose to be consistent in the code.
"To be claimed" means that user will be able to claim it in the future.
"Ready to be claimed" means exactly what it says.
"notYetReadyToBeClaimed" what you have suggested is defined by "isUnstaking"